### PR TITLE
feat(environment): create a test tool that check LSM program support

### PR DESCRIPTION
**Pull request by AlonZivony**
### 1. Explain what the PR does

Create a standalone tool for checking LSM programs support.
This tool is very simple - it tries to load and attach a LSM program to a LSM hook.
If it succeeds and the hook is triggered, it returns true. Else it returns false.
The tool supports both Go usage and Bash usage, making it compatible for both normal usage in Tracee and testing usage.

The tool is loading a kprobe to the same hook before the LSM program test for sanity, promising precise results.

### 2. Explain how to test it

I have tried running the tool in multiple environments to test its support.

### CODE REVIEW

File: common/environment/lsmsupport/README.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+2. Attach the program to the kernel
+3. Trigger the hook by detaching the program (self-triggering)
+4. Check if the hook executed by reading a BPF map
+5. Return definitive support status
+
+## Architecture
+
+```
+lsmsupport/
+├── go.mod                    # Separate module (no libbpfgo dep in main)
```
Comment (yanivagman) [inline common/environment/lsmsupport/README.md]: We don't need or want another go module inside common module

File: common/environment/lsmsupport/check_lsm_support.sh
Comment (yanivagman) [inline common/environment/lsmsupport/check_lsm_support.sh:1]: What do we need this script for?
Isn't the purpose of the go code to check for lsm support?

If this is for test purposes - let's put it under tests and not here

File: common/environment/lsmsupport/cmd/test_cli.go
Comment (yanivagman) [inline common/environment/lsmsupport/cmd/test_cli.go:1]: what do we need this CLI for?
tracee repo isn't a place to host external tools which are out of scope for tracee.

File: common/environment/lsmsupport/go.mod [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module github.com/aquasecurity/tracee/common/environment/lsmsupport
```
Comment (yanivagman) [inline common/environment/lsmsupport/go.mod]: let's remove this module

File: common/environment/lsmsupport/README.md (hunk 1 above)
Comment (AlonZivony) [reply common/environment/lsmsupport/README.md]: Thanks for raising this concern. To clarify my reasoning:
1. This check is intended to be simple and run before any BPF code is loaded. For that reason, I don’t see a strong justification for coupling it with the main BPF logic, so I’ve kept it separate.
2. The logic here is meant to replace the previously merged LSM detection logic, which became too complex to extend whenever new edge cases appeared. Since we agreed that LSM detection belongs under the `common/environment` package, I placed it there.
3. The test requires `libbpfgo`, which would complicate running unit tests in the module if kept inside. Because the current `common` module does not depend on `libbpfgo`, I thought keeping this separately would reduce friction.
4. This module introduces only a single `go.mod` dependency, which makes it straightforward to maintain.

Does this approach address your concern, or would you prefer a different direction?

File: common/environment/lsmsupport/check_lsm_support.sh
Comment (AlonZivony) [reply common/environment/lsmsupport/check_lsm_support.sh:1]: It is a wraparound to the check which make it accessible from the command line.
It is not a proper test, so I am not sure if the test package is the right place to put it under.
Do you think under the script directory would be a good place?

File: common/environment/lsmsupport/cmd/test_cli.go
Comment (AlonZivony) [reply common/environment/lsmsupport/cmd/test_cli.go:1]: This CLI makes it accessible for running it directly with Go.
We will need to run this code in the test suite as we will need to know what to expect when running LSM test - whether it should pass or fail.
It is related to the discussion over the bash script, so I will address this the same - we can put it under the script directory. I just thought that putting it there makes more sense, but I am open to suggestions.

File: common/environment/lsmsupport/check_lsm_support.sh
Comment (AlonZivony) [reply common/environment/lsmsupport/check_lsm_support.sh:1]: I will add that I used this script to check the logic in different environments so it was very useful for me.
As running eBPF code is not trivial, I think it is very useful to have this script that handles all the environment variables needed.

File: builder/Makefile.checkers [hunk 3]
```diff
@@ -98,7 +98,7 @@ help:
 # check formatting (clang-format, goimports-reviser, gofmt)
 #
 
-C_FILES_TO_BE_CHECKED = $(shell find ./pkg/ebpf/c/ -regextype posix-extended -regex '.*\.(h|c)' | xargs)
+C_FILES_TO_BE_CHECKED = $(shell find ./pkg/ebpf/ -regextype posix-extended -regex '.*\.(h|c)' | xargs)
```
Comment (yanivagman) [inline builder/Makefile.checkers]: c files should be in their own directory. Let's keep it this way and move the bpf programs you added there

File: docs/docs/advanced/lsm.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+- **ARM64**: Linux 6.4+ (critical - earlier versions don't work)
+
+### Configuration
+Your kernel needs:
+```bash
+CONFIG_BPF_LSM=y               # BPF LSM support
+CONFIG_LSM="...,bpf"           # Include bpf in LSM list
+```
+
+`CONFIG_LSM` does not include the bpf module for most popular distros by default.
```
Comment (yanivagman) [inline docs/docs/advanced/lsm.md]: this may be true now, but change soon

File: docs/docs/advanced/lsm.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+**Distribution support:**
+
+Note: Most distributions support LSM infrastructure but **BPF LSM is rarely enabled** in default configurations.
+
+- Amazon Linux 2: ❌ Not supported (kernel 5.10, lacks ARM64 infrastructure)
+- Ubuntu 22.04 default: ❌ Not supported (kernel 5.15, lacks ARM64 infrastructure) 
+- Ubuntu 22.04 HWE: ✅ Kernel supports it (6.5+)
+- RHEL 9: ✅ Kernel supports it (backported)
+- Debian 12: ✅ Kernel supports it (6.1+)
```
Comment (yanivagman) [inline docs/docs/advanced/lsm.md]: consider removing

File: docs/docs/advanced/lsm.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+Kernel support works reliably on all modern distributions with kernel 5.7+. However, **BPF LSM is rarely enabled** in standard configurations.
+
+## Why It Might Not Work
+
+**Common reasons LSM BPF fails:**
+
+1. **Kernel too old** (especially ARM64 < 6.4)
+2. **CONFIG_BPF_LSM not enabled** in kernel
+3. **Missing boot parameter** (`lsm=...,bpf`)
+4. **Architecture limitations** (ARM64 on older kernels)
```
Comment (yanivagman) [inline docs/docs/advanced/lsm.md]: repeats first point

File: docs/docs/advanced/lsm.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+## Bottom Line
+
+LSM BPF is a **security policy enforcement** feature that allows custom security rules. Tracee works fine without it. 
+
+**Most systems support LSM infrastructure but BPF LSM is rarely enabled in standard configurations** - this is normal and expected.
+
+- **x86_64**: Kernel support widely available (5.7+)
+- **ARM64**: Kernel support requires 6.4+
+- **Containers/Cloud**: Configuration varies by provider and setup
```
Comment (yanivagman) [inline docs/docs/advanced/lsm.md]: consider removing

File: pkg/ebpf/lsmsupport/lsm_support.go [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+	"sync"
+	"unsafe"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+
+	"github.com/aquasecurity/tracee/common/logger"
+)
+
+var (
+	lsmSupportCache    *bool
```
Comment (yanivagman) [inline pkg/ebpf/lsmsupport/lsm_support.go]: what is the idea of making this pointer to bool and not bool?

File: pkg/ebpf/lsmsupport/lsm_support.go [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+// It first tests kprobe BPF functionality as a sanity check, then tests LSM BPF support.
+// This function is called only once by IsLsmBpfSupported thanks to sync.Once caching.
+func isLsmBpfSupportedImpl() (bool, error) {
+	// Setup libbpf logging to suppress verbose debug output
+	setupLibbpfLogging()
+
+	// Check if we're running as root (required for BPF operations)
+	if os.Geteuid() != 0 {
+		return false, errors.New("LSM BPF test requires root privileges")
+	}
```
Comment (yanivagman) [inline pkg/ebpf/lsmsupport/lsm_support.go]: nit: better make the `Getuid` check first

File: scripts/lsm_test_cli.go
Comment (yanivagman) [inline scripts/lsm_test_cli.go:1]: a go file shouldn't be under scrtips.
If this is a CLI tool, better place it under cmd

File: scripts/check_lsm_support.sh
Comment (yanivagman) [inline scripts/check_lsm_support.sh:1]: this script builds the bpf objects, libbpf and lsm cli tool.
Instead, I think it will be better to make the lsm test cli tool the interface for the user, and put it under cmd.
Then the user can simply run an already compiled go program that just works.
